src/main/java/org/jabref/logic/importer/util/FileFieldParser.java

```diff
@@ -149,32 +149,33 @@ static LinkedFile convert(List<String> entry) {
         LinkedFile field = null;
         if (LinkedFile.isOnlineLink(entry.get(1))) {
             try {
-                field = new LinkedFile(entry.getFirst(), new URL(entry.get(1)), entry.get(2));
+                field = new LinkedFile(entry.get(0), new URL(entry.get(1)), entry.get(2));
             } catch (MalformedURLException e) {
                 // in case the URL is malformed, store it nevertheless
-                field = new LinkedFile(entry.getFirst(), entry.get(1), entry.get(2));
+                field = new LinkedFile(entry.get(0), entry.get(1), entry.get(2));
             }
-        }
-
-        if (field == null) {
+        } else {
```

Comment:
Sure.
The original code has an `if` that checks if the file is an online link. If it is, then it either creates a LinkedFile with a `new URL`, or it creates a LinkedFile with just the string, in case that string can't be converted to a URL and a MalformedURLException is thrown.
Either way, when the `if` is exited, `field` should not be null, so it will never enter the next `if`, so it can be replaced with an `else`